packages/expo-router/src/layouts/StackClient.tsx

```diff
@@ -344,6 +371,31 @@ const Stack = Object.assign(
   }
 );
 
+function disableAnimationInScreenOptions(
+  options: ComponentProps<typeof RNStack>['screenOptions'] | undefined
```

Comment:
Refactor these props to make this a bit easier to read, e.g. `NativeStackScreenOptions`